src/c++/perf_analyzer/client_backend/triton_local/triton_local_client_backend.cc

```diff
@@ -0,0 +1,620 @@
+// Copyright (c) 2020-2021, NVIDIA CORPORATION. All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions
+// are met:
+//  * Redistributions of source code must retain the above copyright
+//    notice, this list of conditions and the following disclaimer.
+//  * Redistributions in binary form must reproduce the above copyright
+//    notice, this list of conditions and the following disclaimer in the
+//    documentation and/or other materials provided with the distribution.
+//  * Neither the name of NVIDIA CORPORATION nor the names of its
+//    contributors may be used to endorse or promote products derived
+//    from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS ``AS IS'' AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED.  IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+// OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "src/clients/c++/perf_analyzer/client_backend/triton_local/triton_local_client_backend.h"
+#include "src/clients/c++/examples/json_utils.h"
+#include "src/clients/c++/perf_analyzer/c_api_helpers/triton_loader.h"
+
+namespace perfanalyzer { namespace clientbackend {
+
+//==============================================================================
+
+Error
+TritonLocalClientBackend::Create(
+    const std::string& url, const ProtocolType protocol,
+    const grpc_compression_algorithm compression_algorithm,
+    std::shared_ptr<Headers> http_headers, const bool verbose,
+    const std::shared_ptr<TritonLoader>& loader,
+    std::unique_ptr<ClientBackend>* client_backend)
+{
+  std::cout << "backend creating" << std::endl;
+  std::unique_ptr<TritonLocalClientBackend> triton_client_backend(
+      new TritonLocalClientBackend(
+          protocol, compression_algorithm, http_headers));
+  if (protocol == ProtocolType::HTTP) {
```

Comment:
I don't understand why we need this in C API version. We must avoid overly relying on our client library for directly communicating with C_API. We must ensure the pipeline is as lite as possible.